### DISCUSSION TITLE
Error in defining custom experiment.

### DISCUSSION THREAD

**Discussion by kumaryash7** [pybamm-team/PyBaMM#4024] [Q&A]
I am trying to define a custom experiment where I am trying to use Discharge capacity as a termination condition for my experiment.
The part of my code implementing this is : -

import pybamm

model = pybamm.lithium_ion.DFN()

model.variables["Capac [A.h]"] = model.variables['Discharge capacity [A.h]']
parameter_values = pybamm.ParameterValues("Prada2013")

def cap_cutoff(variables):
    return variables["Capac [A.h]"] - 0.46

# The CustomTermination class takes a name and function
cap_termination = pybamm.step.CustomTermination(
    name="Capacity cut-off [A.h]", event_function=cap_cutoff
)
#  Provide a list of termination events, each step will stop whenever the first
# termination event is reached
terminations = [cap_termination]
experiment = pybamm.Experiment(
    [
        (
            pybamm.step.c_rate(1, termination=terminations),
            
            "Discharge at 1 C until 2V"
​
        )
        
    ]
)

sim = pybamm.Simulation(model, parameter_values=parameter_values, experiment=experiment)
​
sim.solve(initial_soc=1)

This is giving error as : - 
ModelError: 
                    No key set for variable 'Discharge capacity [A.h]'. Make sure it is included in either
                    model.rhs or model.algebraic in an unmodified form
                    (e.g. not Broadcasted).

**Comment by valentinsulzer** [comment]
This is a bug, it's fixed in develop https://github.com/pybamm-team/PyBaMM/issues/4018
Will be fixed in next release. 
For now if you add another variable `model.variables["double capacity"] = 2 * model.variables["Discharge capacity [A.h]"]` it should work

**Reply by kumaryash7** [reply]
I tried adding this, but an error comes up. I am trying to keep a termination condition of discharge capacity on my discharge curve.
This is the code.

import pybamm

# Set up model and parameters
model = pybamm.lithium_ion.DFN()
# sorted(model.variables.keys())
# model.variable_names()
model.variables["Capac [A.h]"] = model.variables['Discharge capacity [A.h]']
parameter_values = pybamm.ParameterValues("Prada2013")
model.variables["double capacity"] = 2 * model.variables["Discharge capacity [A.h]"]

def cap_cutoff(variables):
    return variables["Capac [A.h]"] - 1
# The CustomTermination class takes a name and function
cap_termination = pybamm.step.CustomTermination(
    name="Capacity cut-off [A.h]", event_function=cap_cutoff
)
#  Provide a list of termination events, each step will stop whenever the first
# termination event is reached
terminations = [cap_termination]
​
experiment = pybamm.Experiment(
    [
        (
            pybamm.step.c_rate(1, termination=terminations),
            
        )
        
    ]
    *1
)
1
# Set up simulation
sim = pybamm.Simulation(model, parameter_values=parameter_values, experiment=experiment)
​
sim.solve(initial_soc=1)
---------------------------------------------------------------------------
SolverError                               Traceback (most recent call last)
Cell In[102], line 4
      1 # Set up simulation
      2 sim = pybamm.Simulation(model, parameter_values=parameter_values, experiment=experiment)
----> 4 sim.solve(initial_soc=1)

File ~/anaconda3/lib/python3.11/site-packages/pybamm/simulation.py:882, in Simulation.solve(self, t_eval, solver, check_model, save_at_cycles, calc_esoh, starting_solution, initial_soc, callbacks, showprogress, **kwargs)
    880 if all(isinstance(step, pybamm.EmptySolution) for step in steps):
    881     if len(steps) == 1:
--> 882         raise pybamm.SolverError(
    883             f"Step '{op_conds_str}' is infeasible "
    884             "due to exceeded bounds at initial conditions. "
    885             "If this step is part of a longer cycle, "
    886             "round brackets should be used to indicate this, "
    887             "e.g.:\n pybamm.Experiment([(\n"
    888             "\tDischarge at C/5 for 10 hours or until 3.3 V,\n"
    889             "\tCharge at 1 A until 4.1 V,\n"
    890             "\tHold at 4.1 V until 10 mA\n"
    891             "])"
    892         )
    893     else:
    894         this_cycle = self.experiment.operating_conditions_cycles[
    895             cycle_num - 1
    896         ]

SolverError: Step '_Step(C-rate, 1, termination=[<pybamm.experiment.step.step_termination.CustomTermination object at 0x179cee510>])' is infeasible due to exceeded bounds at initial conditions. If this step is part of a longer cycle, round brackets should be used to indicate this, e.g.:
 pybamm.Experiment([(
	Discharge at C/5 for 10 hours or until 3.3 V,
	Charge at 1 A until 4.1 V,
	Hold at 4.1 V until 10 mA
])

**Reply by Thiyaga2025** [reply]
@kumaryash7 ,

I am trying similar thing with no success. are you able to overcome this error. will appreciate any help here.

regards
